### PR TITLE
vochain: support CA voting

### PR DESCRIPTION
**Pull request by p4u**
A new census origin OFF_CHAIN_CA is now supported by Vochain.

Simplifying the CA census proof looks like this:

```
proof {
  CAbundle:{"nonce": "0x12345","voterAddress": "0xabcd1234" }
  CAsignature: 0x1234567890
}
```

The CAsignature on CAbundle bytes must compute the Ethereum address
for the CA choosen as censusRoot.

Signed-off-by: p4u <pau@dabax.net>

### CODE REVIEW

File: vochain/apputils.go [hunk 1]
```diff
@@ -30,7 +31,7 @@ import (
 // checkProof checks the validity of a census proof (depending on the origin).
 // key is the data to be proof in behalf the censusRoot.
 // In case of weighted proof, this function will return the weight as second parameter.
-func checkProof(proof *models.Proof, censusOrigin models.CensusOrigin, censusRoot, key []byte) (bool, *big.Int, error) {
+func checkProof(proof *models.Proof, censusOrigin models.CensusOrigin, censusRoot, processID, key []byte) (bool, *big.Int, error) {
```
Comment (jordipainan) [inline vochain/apputils.go:34]: Where is this processID used ?

File: vochain/transaction.go [hunk 2]
```diff
@@ -230,11 +232,11 @@ func VoteTxCheck(vtx *models.Tx, state *State, txID [32]byte, forCommit bool) (*
 					return nil, fmt.Errorf("cannot digest public key: (%w)", err)
 				}
 
-				// check merkle proof
+				// check census proof
 				var valid bool
-				valid, vp.Weight, err = checkProof(tx.Proof, process.CensusOrigin, process.CensusRoot, vp.PubKeyDigest)
+				valid, vp.Weight, err = checkProof(tx.Proof, process.CensusOrigin, process.CensusRoot, process.ProcessId, vp.PubKeyDigest)
```
Comment (jordipainan) [inline vochain/transaction.go:237]: If I'm not wrong and the processID is not used inside the `checkProof` func the first comment also applies here.

File: vochain/apputils.go (hunk 1 above)
Comment (p4u) [reply vochain/apputils.go:34]: This will be used when ECDSA_SALTED signature type is implemented.

File: vochain/transaction.go (hunk 2 above)
Comment (p4u) [reply vochain/transaction.go:237]: Same reply.

**Review by jordipainan — APPROVED** [review]
LGTM

**Pull request merged**